### PR TITLE
[ICDS] UI for THR report

### PR DESCRIPTION
**Pull request by panprosiak**
Hi @calellowitz @Rohit25negi 

I have implemented UI for THR reports according to requirements given in the ticket (https://dimagi-dev.atlassian.net/browse/ICDS-671). Now if user has access to all features they can choose 'Take Home Ratio (THR)' Program area, where 'View by' field is disabled and date starts from July 2019. I have also covered new functionalities with tests.

Need QA: no
Environment: ICDS
Locally tested: yes

Regards,
Maciej

### CODE REVIEW

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 1]
```diff
@@ -124,6 +124,9 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         {id: 8, name: 'AWW Performance Report'},
         {id: 9, name: 'LS Performance Report'},
     ];
+    if (vm.haveAccessToFeatures) {
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/download/download.directive.js]: have you tested this that it does not show when feature flag is off?

File: custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js [hunk 2]
```diff
@@ -285,9 +330,97 @@ describe('Download Directive', function () {
...
+
+        it('tests onIndicatorSelect - child beneficiary list not selected, take home ratio report not selected',
+            function () {
+                var clock = sinon.useFakeTimers(new Date(2018, 8, 1));
+                var date = new Date();
+                var currentYear = date.getFullYear();
+                var currentMonth = date.getMonth() + 1;
+                var yearToFilter = currentYear;
+                controller.selectedIndicator = 5;
+                var existingMonths = moment.months();
```
[the remark this answers is not in the document]
Comment (Rohit25negi) [reply custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js]: Looks relevant

**Comment by Rohit25negi** [comment]
@MaciejChoromanski  can you put the screenshot of the different filter under this reports. Specially: Is it working on National level or not

what are the parameters it sends when different filters are selected under this report?
1. aggregation_level
2. location_id
3. indicator?

also did you tested it on various filters locally?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Hey, Left some comments. Please have a look

File: custom/icds_reports/static/js/directives/download/download.directive.js (hunk 1 above)
Comment (MaciejChoromanski) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: Hi, it was manually tested, I didn't have to write another test, one checking this behavior was already written, it's in 'Download Directive dont have access to features', and looks like this:

it('tests that all users have access to ISSNIP monthly register', function () {
    var length = controller.indicators.length;
    assert.equal(9, length);
});

Regards,
Maciej

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: Yeah, I am talking about manual tests. If it's tested manually then it's fine.

**Comment by Rohit25negi** [comment]
> @MaciejChoromanski can you put the screenshot of the different filter under this reports. Specially: Is it working on National level or not
> 
> what are the parameters it sends when different filters are selected under this report?
> 
> 1. aggregation_level
> 2. location_id
> 3. indicator?
> 
> also did you tested it on various filters locally?

@MaciejChoromanski  ping!

**Comment by MaciejChoromanski** [comment]
@Rohit25negi sorry to keep you waiting, but I've had some Github related issues. 
Here is gif file, which presents different filters in THR report: 
UI_for_THR

Parameters sent when this report is selected:
1. aggregation_level: 5,
2. location_id: depends on selected location,
3. indicator: 10.

Yes, I've tested it locally on different filters.

I hope this is enough information, if I've missed something or misunderstood you, please ping me.

Regards,
Maciej

**Comment by Rohit25negi** [comment]
hi @MaciejChoromanski Thanks for putting out the GIF. looks best way of showing whats happening. This looks mostly correct. Just a couple of things:
1. For this report, user should be able to select only till supervisor level not till awc level
2. aggregation_level: it should be based on what level of location is selected as it depends how many columns will be displayed and also decides other filters: National_level: 0, state_level:1, district_level:2, block_level:3, supervisor_level:4

Can you take care of these? Also can you confirm this report is not available when feature flag is turned off?

**Comment by MaciejChoromanski** [comment]
@Rohit25negi Yes, I will take care of this, also, I confirm that this report is not available when feature flag is turned off.

**Comment by Rohit25negi** [comment]
> @Rohit25negi Yes, I will take care of this, also, I confirm that this report is not available when feature flag is turned off.

awesome. please ping once you are done with the changes

**Comment by MaciejChoromanski** [comment]
@Rohit25negi it may take some time, my laptop stopped working and now I've got a new one, I need to set up whole project once more

**Pull request closed without merging**